CONTRIBUTING.md

```diff
@@ -76,4 +76,101 @@ the proper formatting.
+
+#### Creating a regression test
+
+Each regression test should have its own folder in /tst/regression/test_suites. Lets assume
+we want to add a test which we will call foo_test. We will begin by adding a folder named
+foo_test to /tst/regression/test_suites. Within that folder there must exist at least two files.
+
+1. test_suites/foo_test/\__init__.py
+2. test_suites/foo_test/foo_test.py
+
+The \__init__.py file is left empty, it is used to notify python that it is allowed to
+import the contents from any python file present in the foo_test folder. 
+
+The second file foo_test.py **must** have the same name as the folder it is located in. The
+foo_test.py folder **must** contain an Analyze function. The analyze function is responsible for
+checking the output of the driver (foo_driver) and ensuring it has passed. It is called once the 
+foo_driver has been executed. 
+
+All regression tests are run using the `run_test.py` script located in /test/regression. The reason
+for the python script is to ensure that after the test has been run it meets the appropriate 
+test criteria. 
+
+The `run_test.py` **requires** three different files be specified:
+
+1. the driver - the binary that is to be executed for the test
+2. the input deck - parthenon input file with run parameters
+3. the test folder - the folder in parthenon/tst/regression/test_suites containing test files 
+
+```
+./run_test.py --test_dir test_suites/foo_test --driver location_of_foo_driver  --driver_input location_of_foo_input_deck
+```
+
+Assuming parthenon was compiled in parthenon/build one could run the calculate_pi example with the python script `run_test.py` from /tst/regression using
+
+```
+./run_test.py --test_dir test_suites/calculate_pi --driver ../../build/example/calculate_pi/parthenon-example --driver_input test_suites/calculate_pi/parthinput.regression
+```
+
+More options can be passed to `run_test.py` but those three **must** be specified. For options
+that are supported you can call `run_test.py -h`
+
+Note that running more than a single test with `run_test.py` is intentinally not supported, Running
+groups of tests will be handled by ctest. 
+
+#### Integrating the regression test with CMake
+
+At this point CMake and ctest know nothing about your dandy new regression test. To integrate the
+test with cmake it needs to be added to the CMakeLists.txt file in parthenon/tst/regression. 
+Essentially, the command for running `run_test.py` must be called from within cmake. 
+
+```
+list(APPEND TEST_DIRS foo_test)
+list(APPEND TEST_ARGS "--driver location_of_foo_driver --driver_input Location_of_foo_input_deck")
+```
+
+For the calculate pi example shown above this consists of adding parameters to two cmake lists
+
+```
+list(APPEND TEST_DIRS calculate_pi)
+list(APPEND TEST_ARGS "--driver ${CMAKE_BINARY_DIR}/example/calculate_pi/parthenon-example --driver_input ${CMAKE_CURRENT_SOURCE_DIR}/test_suites/calculate_pi/parthinput.regression")
+```
```

Comment:
How would you implement globbing in this case? There is no guarantee that the driver, or input deck will be located in the same general locations.

Reply:
We could perhaps enforce more structure. Like put a config file in a specific location, and that file specifies driver, input deck, etc locations? Something like that? Then we could glob for those config files?

Heck, the `cmake` commands could be globbed for and included and the cmake can hardcode things? Similar to what @AndrewGaspar did in #114 ?

Reply:
Instead of having everything in one cmake file we would that information in a config file or multiple config files. I guess I'm not really seeing what the advantage is?

Reply:
I guess the advantage I see is that you could have each configuration feel more... self-contained and modular? It feels more natural and manageable to add a new file, that satisfies a format, than to add new code to an existing file. I'm envisioning that someday we have like 10 regression tests.

I say that since this feels less like code and more like build system infrastructure.  Of course, this could just be personal preference on my part.

Reply:
Ok, I see your point, before doing more work on this point though I will wait to hear if anyone else has a strong opinion about it.

Reply:
@Yurlungur because this seems to fall in the preference category, and I don't want to rework something that is working I'm going to count this issue as resolved. It can be revisited later if needed.

Reply:
That's fine.